x-pack/plugins/infra/public/components/asset_details/asset_details.tsx

```diff
@@ -0,0 +1,228 @@
+  onTabClick,
+  renderedTabsSet,
+  currentTimeRange,
+  hostFlyoutOpen,
+  setHostFlyoutOpen,
+  tabs,
+  showInFlyout,
+  links,
+  showActionsColumn,
+  nodeType = NODE_TYPE,
+}: AssetDetailsProps) => {
+  const { euiTheme } = useEuiTheme();
+  const [selectedTabId, setSelectedTabId] = useState('metadata');
+
+  const onTabSelectClick = (tab: Tab) => {
+    renderedTabsSet.current.add(tab.id); // On a tab click, mark the tab content as allowed to be rendered
+    setSelectedTabId(tab.id);
+  };
+
+  const persistMetadataSearchToUrlState =
+    setHostFlyoutOpen && hostFlyoutOpen
+      ? {
+          metadataSearchUrlState: hostFlyoutOpen.metadataSearch,
+          setMetadataSearchUrlState: setHostFlyoutOpen,
+        }
+      : undefined;
+
+  const tabEntries = tabs.map((tab) => (
+    <EuiTab
+      {...tab}
+      key={tab.id}
+      onClick={() => (onTabClick ? onTabClick(tab) : onTabSelectClick(tab))}
+      isSelected={tab.id === hostFlyoutOpen?.selectedTabId ?? selectedTabId}
+    >
+      {tab.name}
+    </EuiTab>
+  ));
+
+  const linksMapping = {
+    apmServices: (
+      <EuiFlexItem grow={false}>
+        <LinkToApmServices hostName={node.name} apmField={'host.hostname'} />
+      </EuiFlexItem>
+    ),
+    uptime: (
+      <EuiFlexItem
+        grow={false}
+        css={css`
+          margin-right: ${euiTheme.size.l};
+        `}
+      >
+        <LinkToUptime nodeType={nodeType} node={node} />
+      </EuiFlexItem>
+    ),
+  };
+
+  const headerLinks = links?.map((link) => linksMapping[link]);
+
+  if (!showInFlyout) {
+    return (
+      <>
+        <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+          <EuiFlexItem>
+            <EuiTitle size="l">
+              <h1>{node.name}</h1>
+            </EuiTitle>
+          </EuiFlexItem>
+          {links && headerLinks}
+        </EuiFlexGroup>
+        <EuiSpacer size="l" />
+        <EuiTabs
+          css={css`
+            margin-bottom: 25px;
+          `}
+          size="l"
+        >
+          {tabEntries}
+        </EuiTabs>
+        {renderedTabsSet.current.has(FlyoutTabIds.METADATA) && (
+          <div hidden={(hostFlyoutOpen?.selectedTabId ?? selectedTabId) !== FlyoutTabIds.METADATA}>
+            <Metadata
+              currentTimeRange={currentTimeRange}
+              node={node}
+              nodeType={nodeType}
+              showActionsColumn={showActionsColumn}
+              persistMetadataSearchToUrlState={persistMetadataSearchToUrlState}
+            />
+          </div>
+        )}
+        {renderedTabsSet.current.has(FlyoutTabIds.PROCESSES) && (
+          <div hidden={(hostFlyoutOpen?.selectedTabId ?? selectedTabId) !== FlyoutTabIds.PROCESSES}>
+            <Processes
+              node={node}
+              nodeType={nodeType}
+              currentTime={currentTimeRange.to}
+              hostFlyoutOpen={hostFlyoutOpen}
+              setHostFlyoutOpen={setHostFlyoutOpen}
+            />
+          </div>
+        )}
```

Comment:
Great idea! Those ended up staying the same (wasn't sure about the UI at the beginning when I added them) I extracted it in a new component as suggested 👍  We can always split them in case of design changes